cmd/util/ledger/migrations/change_contract_code_migration.go

```diff
@@ -77,64 +72,49 @@ func BurnerAddressForChain(chainID flow.ChainID) flow.Address {
 func SystemContractChanges(chainID flow.ChainID, options SystemContractChangesOptions) []StagedContract {
 	systemContracts := systemcontracts.SystemContractsForChain(chainID)
 
-	serviceAccountAddress := systemContracts.FlowServiceAccount.Address
-	idTableStakingAddress := systemContracts.IDTableStaking.Address
-	clusterQCAddress := systemContracts.ClusterQC.Address
-	dkgAddress := systemContracts.DKG.Address
-	fungibleTokenAddress := systemContracts.FungibleToken.Address
-	flowTokenAddress := systemContracts.FlowToken.Address
-	flowFeesAddress := systemContracts.FlowFees.Address
-	flowStorageFeesAddress := systemContracts.FlowStorageFees.Address
-	viewResolverAddress := systemContracts.ViewResolver.Address
-	metadataViewsAddress := systemContracts.MetadataViews.Address
-	fungibleTokenMetadataViewsAddress := common.Address(fungibleTokenAddress)
-	fungibleTokenSwitchboardAddress := common.Address(fungibleTokenAddress)
-
-	burnerAddress := BurnerAddressForChain(chainID)
-
-	var stakingCollectionAddress common.Address
-	var stakingProxyAddress common.Address
+	env := systemContracts.AsTemplateEnv()
+	env.BurnerAddress = BurnerAddressForChain(chainID).Hex()
 
 	switch chainID {
 	case flow.Mainnet:
-		stakingCollectionAddress = mustHexAddress("0x8d0e87b65159ae63")
-		stakingProxyAddress = mustHexAddress("0x62430cf28c26d095")
+		env.StakingCollectionAddress = "0x8d0e87b65159ae63"
+		env.StakingProxyAddress = "0x62430cf28c26d095"
+		env.ViewResolverAddress = "0x1d7e57aa55817448"
+		env.NonFungibleTokenAddress = "0x1d7e57aa55817448"
+		env.MetadataViewsAddress = "0x1d7e57aa55817448"
 
 	case flow.Testnet:
-		stakingCollectionAddress = mustHexAddress("0x95e019a17d0e23d7")
-		stakingProxyAddress = mustHexAddress("0x7aad92e5a0715d21")
+		env.StakingCollectionAddress = "0x95e019a17d0e23d7"
+		env.StakingProxyAddress = "0x7aad92e5a0715d21"
+		env.ViewResolverAddress = "0x631e88ae7f1d7c20"
+		env.NonFungibleTokenAddress = "0x631e88ae7f1d7c20"
+		env.MetadataViewsAddress = "0x631e88ae7f1d7c20"
 
 	case flow.Emulator, flow.Localnet:
-		stakingCollectionAddress = common.Address(serviceAccountAddress)
-		stakingProxyAddress = common.Address(serviceAccountAddress)
+		env.StakingCollectionAddress = env.ServiceAccountAddress
+		env.StakingProxyAddress = env.ServiceAccountAddress
+		env.ViewResolverAddress = env.ServiceAccountAddress
+		env.NonFungibleTokenAddress = env.ServiceAccountAddress
+		env.MetadataViewsAddress = env.ServiceAccountAddress
```

Comment:
Could `NonFungibleTokenAddress`, `MetadataViewsAddress`, and `ViewResolverAddress` be set in `AsTemplateEnv`? 

There's even a "TODO" comment there noting that this wasn't previously possible, but it should be now, right?

Reply:
Good point! I fixed it in my latest commit